pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,238 @@
+			return err
+		}
+	} else if opts.GoTemplate != "" {
+		t := export.NewTemplate(opts.IO, opts.GoTemplate)
+		j, err := json.Marshal(result.Items)
+		if err != nil {
+			return err
+		}
+		err = t.Execute(bytes.NewReader(j))
+		if err != nil {
+			return err
+		}
+	} else {
+		err := displayResults(opts.IO, result)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func displayResults(io *iostreams.IOStreams, results search.Result) error {
+	cs := io.ColorScheme()
+	tp := utils.NewTablePrinter(io)
+	for _, repo := range results.Items {
+		var tags []string
+		private, _ := repo["private"].(bool)
+		fork, _ := repo["fork"].(bool)
+		archived, _ := repo["archived"].(bool)
+		if private {
+			tags = append(tags, "private")
+		} else {
+			tags = append(tags, "public")
+		}
+		if fork {
+			tags = append(tags, "fork")
+		}
+		if archived {
+			tags = append(tags, "archived")
+		}
+		info := strings.Join(tags, ", ")
+		infoColor := cs.Gray
+		if private {
+			infoColor = cs.Yellow
+		}
+		updatedAt, _ := repo["updated_at"].(string)
+		tp.AddField(repo["full_name"].(string), nil, cs.Bold)
+		description, _ := repo["description"].(string)
+		tp.AddField(text.ReplaceExcessiveWhitespace(description), nil, nil)
+		tp.AddField(info, nil, infoColor)
+		if tp.IsTTY() {
+			t, _ := time.Parse(time.RFC3339, updatedAt)
+			tp.AddField(t.Format(time.RFC822), nil, cs.Gray)
+		} else {
+			tp.AddField(updatedAt, nil, nil)
+		}
+		tp.EndRow()
+	}
+
+	if io.IsStdoutTTY() {
+		header := "No repositories matched your search\n"
+		if len(results.Items) > 0 {
+			header = fmt.Sprintf("Showing %d of %d repositories\n\n", len(results.Items), results.TotalCount)
+		}
+		fmt.Fprintf(io.Out, "\n%s", header)
+	}
+
+	return tp.Render()
+}
+
+func NewSearchQuery() search.Query {
+	return search.Query{
+		Kind:  "repositories",
+		Order: shared.NewParameter("order", "string", "desc", shared.OptsValidator([]string{"asc", "desc"})),
+		Sort:  shared.NewParameter("sort", "string", "best match", shared.OptsValidator([]string{"forks", "help-wanted-issues", "stars", "updated"})),
+		Qualifiers: search.Qualifiers{
+			"Archived":         shared.NewQualifier("archived", "bool", "", shared.BoolValidator()),
+			"Created":          shared.NewQualifier("created", "string", "", shared.DateValidator()),
+			"Followers":        shared.NewQualifier("followers", "string", "", shared.RangeValidator()),
+			"Fork":             shared.NewQualifier("fork", "string", "false", shared.OptsValidator([]string{"false", "true", "only"})),
+			"Forks":            shared.NewQualifier("forks", "string", "", shared.RangeValidator()),
+			"GoodFirstIssues":  shared.NewQualifier("good-first-issues", "string", "", shared.RangeValidator()),
+			"HelpWantedIssues": shared.NewQualifier("help-wanted-issues", "string", "", shared.RangeValidator()),
+			"In":               shared.NewQualifier("in", "stringSlice", "name, description", shared.MultiOptsValidator([]string{"name", "description", "readme"})),
```

Comment:
Do we have to validate input values at all?

We don't seem to provide the user with any added value when it comes to dates, ranges, and enums. It just adds complexity to our code and reduces forward-compatibility: e.g. when GitHub search adds another possible value for the `in:` qualifier, GitHub CLI will prevent using that as value because it only allow-lists `name`, `description`, and `readme`. We could add the known valid values to tab-completions for a certain flag, but I don't see why it's ever an added value to limit the user's input to know values. Displaying an argument error to the user isn't worth limiting their options and sacrificing forward compatibility, IMO.

Reply:
The biggest value from validating input is that the server gives no indication when an invalid qualifier is given, it just ignores it and searches for results as if it didn't exist. To me this is a design flaw with the search mechanism, if a bad qualifier is used it should return an error. I think current behavior leads to confusion about the returned results where users see results that don't make any sense and don't have any indication as to what went wrong. An example of this would be if the user mistyped `readme` as `redme` then they would see unexpected results. 

Additionally, I often have to look up the valid values for some of the search qualifiers and if our users have to open up the browser to do that then they probably should just search in the browser since it is already open. There is an option here to show the user valid options but not validate on them, as you suggested, but that also has the issue of forward-compatibility where the docs get out of sync and need to be updated. To me updating the docs and updating the valid options would both be trivial PRs that even community members could do. I do agree that having to stay on top of new search options/features could be a nuisance. 

Lastly, if the user doesn't want any validation they can provide the raw query string. So in the case where a new valid value for `in:` qualifier is added our users can still use it.